### DISCUSSION TITLE
[Question]: always need to start up backend twice

### DISCUSSION THREAD

**Discussion by smadep** [danny-avila/LibreChat#4817] [Troubleshooting]
### What is your question?

**npm run backend** always runs into mongodb timeout when started first from VS code terminal.

Is that normal behavior or do I miss some configuration step?

Any advice?

Thanks

### More Details

I'm running my fork of latest main branch with no changes.
I'm currently running librechat from code (VS Code, Win11) with mongodb-cs and meilisearch docker containers.

I start frontend with npm run frontend:dev and then backend with npm run backend.

And the first time always backend fails to start up with connection timeout errors to mongodb.
Upon second try it works almost always then?!

### What is the main subject of your question?

Other

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
That is not normal behavior and I constantly run it from VS code.

**Comment by smadep** [comment]
Hmm, this is what I get ...

```
error: KeyvMongo connection error: connection timed out
error: There was an uncaught error: connection timed out
```

I debugged through it and the MONGO_URI is set properly?!

Anything worth investigating in .env or librechat.yaml files?
